megalinter/utils.py

```diff
@@ -129,8 +129,8 @@ def filter_files(
         if filter_regex_exclude_object and filter_regex_exclude_object.search(file):
             continue
 
-        # Skip according to file extension (only if lint_all_other_linter_files is false)
-        if lint_all_other_linters_files is False:
+        # Skip according to file extension (only if lint_all_other_linter_files is false or file_extensions is defined)
+        if lint_all_other_linters_files is False or file_extensions is not None:
```

Comment:
default value of file_extension on a linter class is `[]` --> https://github.com/bdovaz/megalinter/blob/20261a8123a6a76a7e399fe56975e249dc97bbfe/megalinter/Linter.py#L69
So you can make the test `len(file_extensions) > 0` :)

Reply:
Changed!